Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1

```diff
@@ -0,0 +1,152 @@
+            Write-LogInfo "  SSH Port : $($vmData.SSHPort)"
+            Write-LogInfo "  Internal IP : $($vmData.InternalIP)"
+            Write-LogInfo ""
+
+            Add-Content -Value "$($vmData.RoleName)=$($vmData.InternalIP)" -Path $constantsFile
+            Add-Content -Value "$($vmData.RoleName)SecondIP=$($vmData.SecondInternalIP)" -Path $constantsFile
+        }
+
+        if ($null -eq $masterVM) {
+            Throw "PERF-XDP-TX-FORWARD requires at least one VM with RoleName of sender"
+        }
+
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+
+        if ($currentTestData.SetupConfig.Networking -imatch "SRIOV") {
+            $DataPath = "SRIOV"
+        }
+        else {
+            $DataPath = "Synthetic"
+        }
+        # Start XDP Installation
+        $installXDPCommand = @"
+bash ./XDPForwardingTest.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $masterVM.PublicIP -port $masterVM.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $user -password $password -upload -runAsSudo
+        # IF Single core then enable single core only
+        $testJob = Run-LinuxCmd -ip $masterVM.PublicIP -port $masterVM.SSHPort `
+            -username $user -password $password -command "bash ./StartXDPSetup.sh" `
+            -RunInBackground -runAsSudo
+        # Terminate process if ran more than 5 mins
+        # TODO: Check max installation time for other distros when added
+        $timer = 0
+        while ($testJob -and ((Get-Job -Id $testJob).State -eq "Running")) {
+            $currentStatus = Run-LinuxCmd -ip $masterVM.PublicIP -port $masterVM.SSHPort `
+                -username $user -password $password -command "tail -2 ~/xdpConsoleLogs.txt | head -1" -runAsSudo
+            Write-LogInfo "Current Test Status: $currentStatus"
+            Wait-Time -seconds 20
+            $timer += 1
+            if ($timer -gt 30) {
+                Throw "XDPFowardingTest did not stop after 10 mins. Please check xdpConsoleLogs."
+            }
+        }
+
+        $currentState = Run-LinuxCmd -ip $masterVM.PublicIP -port $masterVM.SSHPort `
+            -username $user -password $password -command "cat state.txt" -runAsSudo
+
+        if ($currentState -imatch "TestCompleted") {
+            Write-LogInfo "Test Completed"
+            $testResult = "PASS"
+        }
+        elseif ($currentState -imatch "TestAborted") {
+            Write-LogErr "Test Aborted. Last known status: $currentStatus."
+            $testResult = "ABORTED"
+        }
+        elseif ($currentState -imatch "TestSkipped") {

```

Comment:
seems we have a common method to get the translated result.   Function Get-FinalResultHeader()

Reply:
It is a bit different. Does not display logs which I think is important to notify that the script has ran and Reason behind Success/Failure is Last known status. Please let me know your thoughts on this.